treeherder/webapp/api/hash.py

```diff
@@ -0,0 +1,28 @@
+import logging
+
+from rest_framework import exceptions, viewsets
+from rest_framework.decorators import action
+from rest_framework.response import Response
+from rest_framework.status import HTTP_400_BAD_REQUEST
+
+from treeherder.model.models import Commit
+from treeherder.webapp.api.serializers import HashQuerySerializer
+
+logger = logging.getLogger(__name__)
+
+
+class HashViewSet(viewsets.ViewSet):
+    @action(detail=False)
+    def tocommit(self, request, project):
+        query_params = HashQuerySerializer(data=request.query_params)
+        if not query_params.is_valid():
+            return Response(data=query_params.errors, status=HTTP_400_BAD_REQUEST)
+        newhash = query_params.validated_data["newhash"]
+        basehash = query_params.validated_data["basehash"]
+        newpush = Commit.objects.filter(comments__contains=newhash).first()
+        basepush = Commit.objects.filter(comments__contains=basehash).first()
+        if newpush is None or basepush is None:
+            raise exceptions.ValidationError(
```

Comment:
I wonder if this should be provided to the user in a response? @beatrice-acasandrei what do you think?

Reply:
I think it makes sense to provide a response to the user, which means this validation might need to be moved to `HashQuerySerializer`. Might be worth looking into adding the whole validation logic in just one _validate()_ method, but up to you @Andrej1198.

Reply:
@beatrice-acasandrei I moved the validation for this into the serializer as requested, please let me know if you need anything else from me! Let me know if you have any other issues with my PR.